src/Host.cpp

```diff
@@ -369,6 +369,28 @@ void Host::resetProfile()
     qDebug()<<"resetProfile() DONE";
 }
 
+// Saves profile to disk - does not save items dirty in the editor, however.
+// returns true if successful or false+error message if not
+std::pair<bool, QString> Host::saveProfile()
+{
+    QString directory_xml = QDir::homePath() + "/.config/mudlet/profiles/" + getName() + "/current";
+    QString filename_xml = directory_xml + "/" + QDateTime::currentDateTime().toString("dd-MM-yyyy#hh-mm-ss") + ".xml";
+    QDir dir_xml;
+    if (!dir_xml.exists(directory_xml)) {
+        dir_xml.mkpath(directory_xml);
+    }
+    QFile file_xml(filename_xml);
+    if (file_xml.open(QIODevice::WriteOnly)) {
+        XMLexport writer(this);
+        writer.exportHost(&file_xml);
+        file_xml.close();
+        saveModules(1);
+        return std::make_pair(true, QString());
+    } else {
+        return std::make_pair(false, tr("Couldn't save %1 to %2 because: %3").arg(getName(), filename_xml, file_xml.errorString()));
```

Comment:
:thinking:  `file_xml.errorString()` will come from the Qt libraries - and once we get UI language switching in place __those__ will be subject to __I18n__ but will drop in in that form; `getName()` and `filename_xml` will already be in the local language.  So we have three "values" that need to be placed into the final string to display and the place for each of those will not be known until the language to display the message and the "values" is determined.